app/src/main/java/com/example/task/models/TaskViewModel.kt

```diff
@@ -17,4 +19,11 @@ class TaskViewModel(): ViewModel() {
             DBHolder.db.taskDao().addTask(Task(0,task,true))
         }
     }
+    suspend fun getData() :LiveData<List<Task>>{
+        lateinit var task:LiveData<List<Task>>
+        viewModelScope.launch {
+            task =  DBHolder.db.taskDao().getAllPending()
+        }.join()
+        return task
+    }
```

Comment:
Why are you calling `join()` here?

Reply:
Also, I don't think assigning task would work.
1. Use `MutableLiveData`
2. Set `value` property

Like this:

```kotlin
lateinit var task = MutableLiveData<List<Task>>()
viewModelScope.launch {
        task.value =  DBHolder.db.taskDao().getAllPending()
}
return task
```

Also, rename to `tasks`, since it's multiple tasks, not a single task.

Reply:
Won't work that way. Remember we were returning LiveData<List<Task>> through getAllPending().
So we need to assign the value of getAllPending(), i.e getAllPending().value.
I did that and it's working.